src/lib/app/versions.tsx

```diff
@@ -70,6 +69,46 @@ export async function getLatestAvailableAppVersionForApp(
   );
 }
 
+export async function getLatestAvailableTargetAppVersionForAppVersionUpgradeCandidates(
+  apiClient: MittwaldAPIV2Client,
+  appId: string,
+  baseAppVersionId: string,
+): Promise<AppVersion | undefined> {
+  const versions = await apiClient.app.listUpdateCandidatesForAppversion({
+    appId,
+    baseAppVersionId,
+  });
+  assertStatus(versions, 200);
+  if (versions.data.length === 0) {
+    return undefined;
+  }
+  let latestVersion = "0.0.0";
+  for (const version of versions.data) {
+    if (gt(version.internalVersion, latestVersion)) {
```

Comment:
If we're using semver functionalities anyway: Should we just offer the option for the user to specify a semver selector? Like...

```
$ mw app upgrade <app-id> --target-version="^12.4.0"
$ mw app upgrade <app-id> --target-version="*"
```

Consequently, the `install` commands should then support this, as well.

Could also be implemented as a follow-up PR.

Reply:
I like this.
I'll add it as a todo for this PR, but we'll see if we don't just merge somewhere in between already and make a new PR with less impactful stuff